vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -731,8 +731,24 @@ import { ItemCache } from '@vaadin/vaadin-grid/src/vaadin-grid-data-provider-mix
           }
           updateGridItemsInDomBasedOnCache(items);
         }
+        const cacheAndIndex = grid._cache.getCacheAndIndexByKey(pkey);
+        clearSubcache(cacheAndIndex.cache, cacheAndIndex.scaledIndex, index, length);
```

Comment:
The logic above actually seems to operate on full pages. So if `clear(...)` is called with index 0 and length 60, with pageSize of 50, if would actually delete the cache for pages 0 and 1, which means from index 0 to index _99_. The new logic should (?) align with that and not use the `length` per-se, but instead an `endIndex`, which is rounded upwards by grid.pageSize (`const endIndex = index + updatedPageCount * grid.pageSize;`).

Reply:
Done